Reduce RemoteException noise in logs

### Description

Currently, if a cascading failure happens, a node will log several `RemoteException`, it will hold the cause of the failure from the remote node to add more information, and it'll suggest looking at the remote node for the full trace. This new `RemoteException` instance doesn't need to include the full stack trace, which might cause confusion otherwise. This also alleviates the need for filling stacktraces over and over. An example:

```java
2026-06-18 13:23:52,904 ERROR (non-blocking-thread-ispn-2-1-2) [o.i.i.i.InvocationContextInterceptor] ISPN000136: Error executing command PutKeyValueCommand on Cache 'coordinator-failover.xml', writing keys [WrappedByteArray[\(00 (2 bytes)]]org.infinispan.remoting.RemoteException: ISPN000217: Received exception from ispn-5, see cause for remote stack trace
        at org.infinispan.remoting.transport.ResponseCollectors.wrapRemoteException(ResponseCollectors.java:26)
        at org.infinispan.remoting.transport.impl.MapResponseCollector.addException(MapResponseCollector.java:65)
        at org.infinispan.remoting.transport.impl.MapResponseCollector$IgnoreLeavers.addException(MapResponseCollector.java:103)
        at org.infinispan.remoting.transport.ValidResponseCollector.addResponse(ValidResponseCollector.java:29)
        at org.infinispan.remoting.transport.ValidResponseCollector.addResponse(ValidResponseCollector.java:21)
        at org.infinispan.remoting.transport.impl.MultiTargetRequest.actualOnResponse(MultiTargetRequest.java:88)
        at org.infinispan.remoting.transport.impl.ExclusiveTargetRequest.onResponse(ExclusiveTargetRequest.java:55)
        at org.infinispan.remoting.transport.impl.ExclusiveTargetRequest.onResponse(ExclusiveTargetRequest.java:30)
        at org.infinispan.remoting.transport.impl.RequestRepository.addResponse(RequestRepository.java:50)
        at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processResponse(JGroupsTransport.java:1575)
        at org.infinispan.remoting.transport.jgroups.JGroupsTransport.processMessage(JGroupsTransport.java:1471)
        at org.infinispan.remoting.transport.jgroups.JGroupsTransport$ChannelCallbacks.up(JGroupsTransport.java:1641)
        at org.jgroups.JChannel.up(JChannel.java:678)
        at org.jgroups.stack.ProtocolStack.up(ProtocolStack.java:951)
        at org.jgroups.protocols.FRAG2.up(FRAG2.java:136)
        at org.jgroups.protocols.FlowControl.up(FlowControl.java:244)
        at org.jgroups.protocols.FlowControl.up(FlowControl.java:244)
        at org.jgroups.protocols.pbcast.GMS.up(GMS.java:796)
        at org.jgroups.protocols.pbcast.STABLE.up(STABLE.java:234)
        at org.jgroups.protocols.UNICAST3.deliverMessage(UNICAST3.java:1172)
        at org.jgroups.protocols.UNICAST3.addMessage(UNICAST3.java:898)
        at org.jgroups.protocols.UNICAST3.handleDataReceived(UNICAST3.java:880)
        at org.jgroups.protocols.UNICAST3.up(UNICAST3.java:490)
        at org.jgroups.protocols.pbcast.NAKACK2.up(NAKACK2.java:644)
        at org.jgroups.protocols.VERIFY_SUSPECT2.up(VERIFY_SUSPECT2.java:104)
        at org.jgroups.protocols.FailureDetection.up(FailureDetection.java:182)
        at org.jgroups.protocols.FD_SOCK2.up(FD_SOCK2.java:191)
        at org.jgroups.protocols.MERGE3.up(MERGE3.java:271)
        at org.jgroups.protocols.Discovery.up(Discovery.java:304)
        at org.jgroups.stack.Protocol.up(Protocol.java:365)
        at org.jgroups.protocols.TP.passMessageUp(TP.java:426)
        at org.jgroups.util.SubmitToThreadPool$SingleMessageHandler.run(SubmitToThreadPool.java:99)
        at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(ThreadPerTaskExecutor.java:291)
        at java.base/java.lang.VirtualThread.run(VirtualThread.java:460)
Caused by: org.infinispan.commons.TimeoutException: ISPN000726: Unable to acquire lock for WrappedByteArray[\(00 (2 bytes)] by requestor GlobalTransaction{id=28932, addr=ispn-2, remote=true, xid=null, internalId=-1}. Timeout of 10 seconds will likely fail
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
        at org.infinispan.marshall.protostream.impl.MarshallableThrowable.getInstance(MarshallableThrowable.java:119)
        at org.infinispan.marshall.protostream.impl.MarshallableThrowable.lambda$recreateGenericThrowable$1(MarshallableThrowable.java:93)
        at org.infinispan.marshall.protostream.impl.MarshallableThrowable.create(MarshallableThrowable.java:111)
        at org.infinispan.marshall.protostream.impl.MarshallableThrowable.recreateGenericThrowable(MarshallableThrowable.java:93)
        at org.infinispan.marshall.protostream.impl.MarshallableThrowable.get(MarshallableThrowable.java:78)
        at org.infinispan.remoting.responses.ExceptionResponse.getException(ExceptionResponse.java:31)
        ... 31 more
```

The `cause` includes the more useful information; the remaining stack trace is just noise.

### Implementation ideas

Utilize a `@Param` argument when throwing the `RemoteException` so we can customize whether to include the stack trace.

--- a/core/src/main/java/org/infinispan/remoting/RemoteException.java
+++ b/core/src/main/java/org/infinispan/remoting/RemoteException.java
@@ -14,4 +14,8 @@ public class RemoteException extends CacheException {
       super(msg, cause);
    }
 
+   public RemoteException(String msg, Throwable cause, boolean writableStackTrace) {
+      super(msg, cause, true, writableStackTrace);
+   }
+
 }

--- a/core/src/main/java/org/infinispan/remoting/transport/ResponseCollectors.java
+++ b/core/src/main/java/org/infinispan/remoting/transport/ResponseCollectors.java
@@ -23,7 +23,7 @@ public class ResponseCollectors {
          e = (CacheException) exception;
       } else {
          // if we have any application-level exceptions make sure we throw them!!
-         e = CLUSTER.remoteException(sender, exception);
+         e = CLUSTER.remoteException(sender, exception, false);
       }
       return e;
    }

--- a/core/src/main/java/org/infinispan/topology/TopologyManagementHelper.java
+++ b/core/src/main/java/org/infinispan/topology/TopologyManagementHelper.java
@@ -136,7 +136,7 @@ public class TopologyManagementHelper {
       Response localResponse;
       if (t != null) {
          localResponse = new ExceptionResponse(
-               CLUSTER.remoteException(localAddress, CompletableFutures.extractException(t)));
+               CLUSTER.remoteException(localAddress, CompletableFutures.extractException(t), true));
       } else {
          if (v instanceof Response) {
             localResponse = ((Response) v);

--- a/core/src/main/java/org/infinispan/util/logging/Log.java
+++ b/core/src/main/java/org/infinispan/util/logging/Log.java
@@ -820,7 +820,7 @@ public interface Log extends BasicLogger {
    void entriesMigrated(long count, String name, String prettyTime);
 
    @Message(value = "Received exception from %s, see cause for remote stack trace", id = 217)
-   RemoteException remoteException(Object sender, @Cause Throwable t);
+   RemoteException remoteException(Object sender, @Cause Throwable t, @Param boolean writableStackTrace);
 
 //   @LogMessage(level = INFO)
 //   @Message(value = "Timeout while waiting for the transaction validation. The command will not be processed. " +

